src/service/viewport-impl.js

```diff
@@ -1022,12 +1047,17 @@ export class ViewportBindingNaturalIosEmbed_ {
   }
 
   /** @override */
-  updatePaddingTop(paddingTop) {
+  updatePaddingTop(paddingTop, opt_updateScrollPos, opt_lastPaddingTop) {
     onDocumentReady(this.win.document, doc => {
       this.paddingTop_ = paddingTop;
       // Also tried `paddingTop` but it didn't work for `position:absolute`
       // on iOS.
       doc.body.style.borderTop = `${paddingTop}px solid transparent`;
+      if (opt_updateScrollPos) {
+        const adjScrollTop = this.getScrollTop() + 3 * paddingTop
```

Comment:
Add a comment to explain 3x, e.g. "compensate for padding being removed by the scroller and removed border."